git/squash.sh

```
@@ -0,0 +1,38 @@
+
+# Switch back to the current branch after updating the parent branch
+echo "Switching back to the current branch '$current_branch'..."
+git checkout $current_branch
+
+# Generate a generic commit message
+commit_message="Squashed commits for '$current_branch' relative to '$parent_branch'"
+
+# Reset current branch to the point where it diverged from the parent branch
+# Then, stage all changes and create a single new commit with a generic commit message
+echo "Resetting and squashing commits..."
+git reset --soft $(git merge-base $current_branch $parent_branch)
```

Shellcheck (0.10.0)

[warning] Quote this to prevent word splitting.

(SC2046)